java/vector/src/main/codegen/templates/ComplexWriters.java

```diff
@@ -191,7 +195,9 @@ public void writeNull() {
 public interface ${eName}Writer extends BaseWriter {
   public void write(${minor.class}Holder h);
 
-  <#if minor.class?starts_with("Decimal")>@Deprecated</#if>
+<#if minor.class?starts_with("Decimal") || is_timestamp_tz(minor.class) || minor.class == "Duration" || minor.class == "FixedSizeBinary">
+  @Deprecated
```

Comment:
Is it possible to generate a basic JavaDoc with a `@deprecated` tag with a little more context?